src/pages/standard-report/HtmlReport.js

```diff
@@ -37,26 +38,67 @@ const Loader = () => (
     </div>
 )
 
-const HtmlReport = ({ html }) =>
-    html ? (
-        <iframe
-            id="html-report-id"
-            srcDoc={wrapHtmlInTemplate(html)}
-            title="html-report-content"
-            width="100%"
-            seamless={true}
-            sandbox="allow-same-origin allow-scripts allow-modals"
-        >
-            <style jsx>{`
-                iframe {
-                    border: none;
-                    flex-grow: 1;
-                }
-            `}</style>
-        </iframe>
-    ) : (
-        <Loader />
-    )
+class HtmlReport extends Component {
+    state = { heigth: 'auto' }
+    iframeHtml = null
+    iframeBody = null
+    heightObserver = null
+
+    onIframeLoad = event => {
+        const iframe = event.target
+
+        this.iframeHtml = iframe.contentWindow.document.documentElement
+        this.iframeBody = iframe.contentWindow.document.body
+        this.heightObserver = new window.ResizeObserver(this.adjustHeight)
+
+        this.heightObserver.observe(this.iframeHtml)
+        this.heightObserver.observe(this.iframeBody)
+        this.adjustHeight()
+    }
+
+    adjustHeight = () => {
+        const height = Math.max(
+            this.iframeBody.clientHeight,
+            this.iframeBody.scrollHeight,
+            this.iframeBody.offsetHeight,
+            this.iframeHtml.clientHeight,
+            this.iframeHtml.scrollHeight,
+            this.iframeHtml.offsetHeight
+        )
+
+        // Add 20px as a "safety margin" in case we get a horizontal scroll bar
+        this.setState({ height: height + 20 })
+    }
+
+    componentWillUnmount() {
```

Comment:
I think you can also use disconnect to unobserve everything at once: https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver/disconnect

Reply:
🎉  I overlooked that method, and since I don't have to pass that one a reference to a node, I could simplify my code quite a bit. Thanks